File: benchmark/results/java-invocation-inline/run-003/b-baseline/output/com/metaengine/demo/catalog/services/ProductRepository.java
package com.metaengine.demo.catalog.services;

import com.metaengine.demo.catalog.aggregates.Product;
import java.util.List;
import java.util.Optional;

/** ProductRepository persistence service for the catalog domain. */
public class ProductRepository {
    public Product create(Product input) {
        throw new UnsupportedOperationException("not implemented");
    }

    public Optional<Product> findById(String id) {
        throw new UnsupportedOperationException("not implemented");
    }

    public List<Product> list(int limit) {
        throw new UnsupportedOperationException("not implemented");
    }

    public void delete(String id) {
        throw new UnsupportedOperationException("not implemented");
    }
}

File: benchmark/results/python-invocation-script/run-003/a-mcp/output/catalog/services/product_repository.py
from catalog.aggregates.product import Product

"""ProductRepository service."""
class ProductRepository:
    def create(self, input: Product) -> Product:
        raise NotImplementedError('not implemented')
    def findById(self, id: str) -> Product | None:
        raise NotImplementedError('not implemented')
    def list(self, limit: float) -> list[Product]:
        raise NotImplementedError('not implemented')
    def delete(self, id: str) -> None:
        raise NotImplementedError('not implemented')

